src/qibolab/instruments/unrolling.py

```diff
@@ -17,6 +17,26 @@ def batch_max_sequences(sequences, max_size):
     return chunked(sequences, max_size)
 
 
+def batch_max_duration(sequences, max_duration):
+    """Split a list of sequences to batches with a trying to get an waveform
+    memory for the control pulses estimate using the duration of the sequence.
+
+    Args:
+        sequences (list): List of :class:`qibolab.pulses.PulseSequence` objects.
+        max_duration (int): Maximum number of readout pulses in a single batch.
+    """
+    batch_duration, batch = 0, []
+    for sequence in sequences:
+        duration = sequence.duration - sequence.ro_pulses.duration
+        if duration + batch_duration > max_duration:
+            yield batch
+            batch_duration, batch = duration, [sequence]
+        else:
+            batch.append(sequence)
+            batch_duration += duration
+    yield batch
```

Comment:
This is already pretty similar to the one below, let me make an alternative proposal.

Reply:
Yes, I wrote the todos for this reason:
https://github.com/qiboteam/qibolab/blob/9d487568718bea3bfd6f3150887e2ddaea35b2a3/src/qibolab/instruments/unrolling.py#L69-L71

But I'm not sure if you should really remove the readout duration: for reading you need a readout pulse, and that pulse has a waveform that has to be uploaded.
Maybe we're reading for more time that what we're playing on readout. But this will be automatically solved by summing the non-rectangular & non-delay durations.

If instead readout is rectangular, it should be stripped as any other rectangular. But if you choose anything else, you will have to upload it.